zerver/webhooks/newrelic/view.py

```diff
@@ -1,169 +1,264 @@
 
-# Once old is EOLed, delete old and keep new
-OLD_EVENT_TYPES = ["closed", "acknowledged", "open"]
-NEW_EVENT_TYPES = ["created", "activated", "acknowledged", "closed"]
-ALL_EVENT_TYPES = list(set(OLD_EVENT_TYPES).union(set(NEW_EVENT_TYPES)))
+    message_context: dict[str, str | None] = {
+        "title": payload.get("title", "No title.").tame(check_string),
+        "conditions_str": conditions_str,
+        "alert_policy_str": alert_policy_str,
+        "time_created_str": TIME_STAMP_STR.get("CREATED"),
+        "time_updated_str": TIME_STAMP_STR.get("UPDATED"),
+        "incident_url": payload.get("issueUrl", DEFAULT_NEWRELIC_URL).tame(check_string),
+        "total_incidents": str(payload["totalIncidents"].tame(check_int)),
+        "state": payload.get("state", "UPDATED").tame(check_string_in(INCIDENT_STATES)),
+        "status": determine_status(payload, TIME_STAMP_STR),
+        "priority": priority,
+        "priority_symbol": priority_symbol,
+    }
 
+    return message_context
 
-@webhook_view("NewRelic", all_event_types=ALL_EVENT_TYPES)
+
+def determine_status(payload: WildValue, time_stamp_map: dict[str, str | None]) -> str:
+    """
+    This function infers what type of action was taken on the incident based on the
+    timestamps provided in the payload.
+
+    Returns the action taken as a string.
+    """
+    updated_at = time_stamp_map.get("UPDATED")
+    acknowledged_at = time_stamp_map.get("ACKNOWLEDGED")
+    created_at = time_stamp_map.get("CREATED")
+    closed_at = time_stamp_map.get("CLOSED")
+
+    if updated_at:
+        if acknowledged_at and updated_at == acknowledged_at:
+            return "ACKNOWLEDGED"
+        elif created_at and updated_at == created_at:
+            return "ACTIVATED"
+        elif closed_at and updated_at == closed_at:
+            return "CLOSED"
+        else:
+            return "UPDATED"
+    incident_title = payload.get("title", "No title.").tame(check_string)
+    raise UnsupportedWebhookEventTypeError(incident_title)
+
+
+def generate_additional_info_message(payload: WildValue) -> str | None:
+    """
+    Parses additional fields from the recommended base payload
+    into a presentable block of strings.
+    Detail: zerver/webhooks/newrelic/doc.md/#recommended-base-payload-template
+
+    Returns a block of formatted strings.
+    """
+    owner = payload.get("owner").tame(check_none_or(check_string))
+    additional_info_str = f"|:silhouette:  Acknowledged by **{owner}**|\n"
+    return None if owner is None else additional_info_str
+
+
+def generate_custom_field_message(payload: WildValue) -> str | None:
+    """
+    Parses the custom fields in the "zulipCustomFields" dict in
+    the payload by turning the camel case key into words and
+    formatting it together with the value into a presentable
+    block of strings.
+    Detail: zerver/webhooks/newrelic/doc.md/#displaying-custom-fields
+
+    Returns a block of formatted strings.
```

Comment:
I don't think we need this documentation on these functions in this integration. Let's just make sure the function names are clear in what they do. For example, here I might revise the function name to be something like `format_zulip_custom_fields`.